### PR TITLE
CreateImageWizard: add google upload step

### PR DESCRIPTION
**Pull request by jkozol**
The user can now specify their authentication settings for Google Cloud Platform.

Tests and validation are still needed for this PR.

emptyGoogle
googleDomain
googlePopover

### CODE REVIEW

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+import { Form, FormGroup, TextList, TextListItem, Popover, Radio, TextContent, Text, TextInput, Title } from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+
+const WizardStepUploadGoogle = (props) => {
+    const accountTypePopover = (
+        <Popover
+            hasAutoWidth
+            maxWidth='35rem'
+            headerContent={ 'Valid types of account emails' }
```
Comment (thozza) [inline src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js:12]: Although 3 out of 4 types are really emails, maybe we could reword this to "Valid account types"?

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            bodyContent={ <TextContent>
+                <Text>The following types of emails can have an image shared with them:</Text>
```
Comment (thozza) [inline src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js:14]: Same feedback here as above. Personally I would use "The following account types ...".

**Review by thozza — APPROVED** [review]
This looks very nice. Great work! I added just two wording suggestions. I can not assess the code correctness, since I have almost no experience with JS...

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js (hunk 1 above)
Comment (jkozol) [reply src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js:12]: Good point! I will change this.

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js (hunk 2 above)
Comment (jkozol) [reply src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js:14]: Will update this too.

**Comment by katierik** [comment]
LGTM! 🎉

File: src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js (hunk 2 above)
Comment (katierik) [reply src/PresentationalComponents/CreateImageWizard/WizardStepUploadGoogle.js:14]: Updated in mocks as well

**Comment by jkozol** [comment]
@Gundersanne This is ready for review. There are a couple minor issues with the PR. The upload state has gotten a little messy and should be refactored in the future. Also there is no validation. I have opened an issue for this https://github.com/osbuild/image-builder-frontend/issues/98

File: src/SmartComponents/CreateImageWizard/CreateImageWizard.js [hunk 3]
```diff
@@ -240,34 +252,40 @@ class CreateImageWizard extends Component {
...
+        if (this.state.uploadDestinations.google) {
+            const upload_google = this.state.uploadGoogle;
+            delete upload_google.accountType;
+            let request = {
+                distribution: this.state.release,
+                image_requests: [
+                    {
+                        architecture: this.state.arch,
+                        image_type: 'gcp',
+                        upload_requests: [ upload_google ],
```
Comment (Gundersanne) [inline src/SmartComponents/CreateImageWizard/CreateImageWizard.js:281]: Right this is getting a little confusing. I think it's best if we don't consider the upload (options) in the state to be directly usable in the request. And build the upload_request here in the onSave:

```
...
upload_requests: [{
    type: 'gcp',
    options: {
        share_with_accounts: upload_google.options.share_with_accounts
    }
}]
...
```

Can do that in a followup though because AWS would need the same treatment. That way we could also just flatten the upload state to just have `upload_google.share_with_accounts`, and not mess with the options all the time.

File: src/SmartComponents/CreateImageWizard/CreateImageWizard.js [hunk 4]
```diff
@@ -54,8 +56,9 @@ class CreateImageWizard extends Component {
             },
             uploadGoogle: {
                 type: 'google',
```
Comment (Gundersanne) [inline src/SmartComponents/CreateImageWizard/CreateImageWizard.js:58]: This should be `type: 'gcp',`

**Review by Gundersanne — CHANGES_REQUESTED** [review]
Looks good so far, the type should be `gcp` though.

File: src/SmartComponents/CreateImageWizard/CreateImageWizard.js (hunk 4 above)
Comment (jkozol) [reply src/SmartComponents/CreateImageWizard/CreateImageWizard.js]: Fixed

**Review by Gundersanne — APPROVED** [review]
This looks good, I also created #100 to refactor the upload states.

I'd wait with merging this until we redeploy a composer version which includes gcp though, like we shouldn't enable things that don't work yet. Until then I'm adding the blocked label, and I'll get on deploying a new composer version now.

File: src/SmartComponents/CreateImageWizard/CreateImageWizard.js (hunk 3 above)
Comment (teg) [reply src/SmartComponents/CreateImageWizard/CreateImageWizard.js:281]: Tracked in #100
